### PR TITLE
[improve] [Follow-PR#3596] Optimize log trace enable to reduce allocating objects

### PR DESCRIPTION
**Pull request by wenbingshen**
### Motivation

Following PR #3596

### CODE REVIEW

**Comment by wenbingshen** [comment]
@hangc0276 PTAL. Thanks!

File: stream/distributedlog/core/src/main/java/org/apache/distributedlog/BKAsyncLogReader.java [hunk 1]
```diff
@@ -364,7 +364,9 @@ private boolean checkClosedOrInError(String operation) {
 
         Throwable cause = lastExceptionUpdater.get(this);
         if (null != cause) {
-            LOG.trace("Cancelling pending reads");
+            if (LOG.isTraceEnabled()) {
```
Comment (nicoloboschi) [inline stream/distributedlog/core/src/main/java/org/apache/distributedlog/BKAsyncLogReader.java:367]: this is not useful

File: stream/distributedlog/core/src/main/java/org/apache/distributedlog/impl/metadata/ZKLogStreamMetadataStore.java [hunk 2]
```diff
@@ -269,10 +269,14 @@ public void processResult(final int rc, final String path, Object ctx, String na
                                             logMetadata.getFullyQualifiedName())));
                         } else if (KeeperException.Code.OK.intValue() == rc) {
                             FutureUtils.complete(promise, null);
-                            LOG.trace("Created path {}.", path);
+                            if (LOG.isTraceEnabled()) {
```
Comment (nicoloboschi) [inline stream/distributedlog/core/src/main/java/org/apache/distributedlog/impl/metadata/ZKLogStreamMetadataStore.java:272]: this is not useful

File: stream/distributedlog/core/src/main/java/org/apache/distributedlog/impl/metadata/ZKLogStreamMetadataStore.java [hunk 3]
```diff
@@ -269,10 +269,14 @@ public void processResult(final int rc, final String path, Object ctx, String na
...
+                                LOG.trace("Created path {}.", path);
+                            }
                         } else if (KeeperException.Code.NODEEXISTS.intValue() == rc) {
                             FutureUtils.complete(promise, null);
-                            LOG.trace("Path {} is already existed.", path);
+                            if (LOG.isTraceEnabled()) {
```
Comment (nicoloboschi) [inline stream/distributedlog/core/src/main/java/org/apache/distributedlog/impl/metadata/ZKLogStreamMetadataStore.java:277]: this is not useful

File: stream/distributedlog/core/src/main/java/org/apache/distributedlog/lock/ZKDistributedLock.java [hunk 4]
```diff
@@ -181,7 +181,9 @@ void doAsyncAcquireWithSemaphore(final CompletableFuture<ZKDistributedLock> acqu
 
     void doAsyncAcquire(final CompletableFuture<ZKDistributedLock> acquirePromise,
                         final long lockTimeout) {
-        LOG.trace("Async Lock Acquire {}", lockPath);
+        if (LOG.isTraceEnabled()) {
```
Comment (nicoloboschi) [inline stream/distributedlog/core/src/main/java/org/apache/distributedlog/lock/ZKDistributedLock.java:184]: this is not useful

File: stream/distributedlog/core/src/main/java/org/apache/distributedlog/lock/ZKSessionLock.java [hunk 5]
```diff
@@ -770,7 +770,9 @@ public void execute() {
 
                                         currentNode = name;
                                         currentId = getLockIdFromPath(currentNode);
-                                        LOG.trace("{} received member id for lock {}", lockId, currentId);
+                                        if (LOG.isTraceEnabled()) {
```
Comment (nicoloboschi) [inline stream/distributedlog/core/src/main/java/org/apache/distributedlog/lock/ZKSessionLock.java:773]: this is not useful

**Review by nicoloboschi — CHANGES_REQUESTED** [review]
If there's no computation to get the value to print, there's no need to add those conditions

File: stream/distributedlog/core/src/main/java/org/apache/distributedlog/BKAsyncLogReader.java (hunk 1 above)
Comment (wenbingshen) [reply stream/distributedlog/core/src/main/java/org/apache/distributedlog/BKAsyncLogReader.java:367]: @nicoloboschi I think we should be consistent in style, consistent in form. What do you think?

Comment (nicoloboschi) [reply stream/distributedlog/core/src/main/java/org/apache/distributedlog/BKAsyncLogReader.java:367]: I think it's fine to leave the logging libraries to perform optimization if needed.
In this case there's no improvement.
Also, the code is less readable

It's fine to have cases where we check if some level is enabled to not perform values computation if not needed but in general it's not a good practice to guard every logging call

**Pull request closed without merging**